### PR TITLE
=act #15865 add LogEvent.getMDC Java API

### LINKED ISSUE #15865: make LoggingAdapterTest compile again under JDK8
At least in Eclipse it does not work, saying that `toString()` is ambiguous on type Map. We should add a Java API that returns a Java Map.

### PR DESCRIPTION
**Pull request by rkuhn**
closes #15865

### CODE REVIEW

File: akka-actor/src/main/scala/akka/event/Logging.scala [hunk 1]
```diff
@@ -632,6 +632,14 @@ object Logging {
...
      */
     def mdc: MDC = emptyMDC
+
+    /**
+     * Java API: Retrieve the contents of the MDC.
+     */
+    def getMDC: java.util.Map[String, Any] = {
+      import scala.collection.JavaConverters._
+      mdc.asJava
+    }
```
Comment (patriknw) [inline akka-actor/src/main/scala/akka/event/Logging.scala]: good one

**Comment by drewhk** [comment]
LGTM, nice catch

File: akka-actor-tests/src/test/java/akka/event/LoggingAdapterTest.java [hunk 2]
```diff
@@ -128,7 +128,7 @@ protected Void match(Object event) {
                     LogEvent log = (LogEvent) event;
                     assertEquals(message, log.message());
                     assertEquals(level, log.level());
-                    assertEquals(mdc, log.mdc().toString());
+                    assertEquals(mdc, log.getMDC().toString());
```
Comment (patriknw) [inline akka-actor-tests/src/test/java/akka/event/LoggingAdapterTest.java]: hm, what is the purpose of this `expectLog` ?
`String mdc` vs `Map.toString`?
The order of the entries of a Map is undefined.

What is the purpose of `LogJavaTestKit`? The methods have package private visibility. They should at least be protected if it is user api.

Comment (rkuhn) [inline akka-actor-tests/src/test/java/akka/event/LoggingAdapterTest.java]: It is only used within this test, and only with MDCs of size one, so this is why it works. The other purpose was to get Eclipse JDT confused about the API to remind me that a Java version is still needed ;-)

**Pull request merged**